### DISCUSSION TITLE
APP Health status stunk on Progress, which may cause by pods which controlled by pod

### DISCUSSION THREAD

**Discussion by ipomoealba** [argoproj/argo-cd#7201] [Q&A]
Hi all,

We use `argo-cd` to manage our spark applications. The application is base on the `spark-operator`[1]. It has its own CRDs and creates pods automatically. 

So we add the following annotation[2][3] in the extraneous resources (spark driver and spark executors, which have the red boxes in the graph). 

```yaml
    annotations:
      "argocd.argoproj.io/sync-options": Prune=false
      "argocd.argoproj.io/compare-options": IgnoreExtraneous
```

![截圖 2021-09-11 下午2 22 59](https://user-images.githubusercontent.com/26810013/132939419-ddf61cd8-195f-4144-b0f8-c8ff7dbcf417.png)

The resource can be created and the `Sync Status` is `synced`. However, `APP Health` always stunk at `Progress`.

![截圖 2021-09-11 下午2 57 23](https://user-images.githubusercontent.com/26810013/132939527-60fd46f7-308d-41be-9498-5f7cd093d88f.png)

Not sure the reason is, but my proposal is: 

> The executors are created by spark driver, which is also an extraneous resource, so the argo-cd cannot get their status correctly. Children's resources status is `unknown`, then the `APP Health` keeps on `Progress`.

Does anyone encounter the same issue or have any idea to resolve it?

Thank you for your time.

BR,
Oscar

---
Reference
* [1] https://github.com/GoogleCloudPlatform/spark-on-k8s-operator
* [2] https://argoproj.github.io/argo-cd/user-guide/compare-options
* [3] https://argoproj.github.io/argo-cd/user-guide/sync-options

**Comment by wtam2018** [comment]
I think you may need to restore the health assessment of Application CRD.   See here https://argoproj.github.io/argo-cd/operator-manual/health/#argocd-app for more information.
